### PR TITLE
Add push notification docs

### PR DESCRIPTION
**Pull request by mmaatttt**
Docs for:
1. Automatic push config
2. Manual push config (also a tutorial formatted doc)
3. Rich push config
4. Push debugging

I put manual config in a separate doc to make the happy path of using automatic config a bit more prominent and to make the doc look a lot less daunting at first glance.

The automatic config doc is listed at the top level with "Core Functionality" and the other detail docs are nested with the "Push Notifications" topic of `Appcues` docs:
Screenshot 2024-07-09 at 1 12 05 PM

We can also decide to move some of this info to docs.appcues.com if we want, but we've gotta start writing it down somewhere.

### CODE REVIEW

File: Sources/AppcuesKit/AppcuesKit.docc/Tutorials/UseCases.tutorial [hunk 1]
```diff
@@ -5,6 +5,14 @@
...
     }
     
+    @Chapter(name: "Manual Push Notifications") {
+        The Appcues iOS SDK supports receiving push notification so you can reach your users whenever the moment is right.
+        
+        @Image(source: PushNotifications, alt: "Push Notifications")
+        
+        @TutorialReference(tutorial: "doc:PushNotificationsTutorial")
+    }
+
```
Comment (iujames) [inline Sources/AppcuesKit/AppcuesKit.docc/Tutorials/UseCases.tutorial:15]: just a note that came to mind - we may want to add one about the push primer flow and request-push action at some point (also reminder to selves to ticket mobile builder work to add that new action when the SDK has shipped). relates somewhat to primer item below, but different

File: Sources/AppcuesKit/AppcuesKit.docc/PushNotificationsRich.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
++ class NotificationService: AppcuesNotificationServiceExtension {}
+- class NotificationService: UNNotificationServiceExtension {
+-     ...
+- }
+```
+
+## Step 4: Update the Apple Developer Portal
+
+A Notification Service Extension is a separate binary that's bundled with your app. It must be set up in the Apple Developer Portal with its own app ID and provisioning profile.
```
Comment (iujames) [inline Sources/AppcuesKit/AppcuesKit.docc/PushNotificationsRich.md:54]: worth trying to link directly to some Apple dev ref docs here or anywhere else?

**Review by iujames — APPROVED** [review]
looking great! love the tutorial and the way things are laid out here to emphasize the automatic route

File: Sources/AppcuesKit/AppcuesKit.docc/Tutorials/UseCases.tutorial (hunk 1 above)
Comment (iujames) [reply Sources/AppcuesKit/AppcuesKit.docc/Tutorials/UseCases.tutorial:15]: added ticket for builder here in the GA objective https://app.shortcut.com/appcues/story/67776/add-request-push-action-in-mobile-builder-for-push-primers

File: Sources/AppcuesKit/AppcuesKit.docc/PushNotificationsRich.md (hunk 2 above)
Comment (mmaatttt) [reply Sources/AppcuesKit/AppcuesKit.docc/PushNotificationsRich.md:54]: It'd be nice to but I'm not aware of an Apple doc that specifically covers this.

**Pull request merged**